test/sql/cte/recursive_cte_key_variant.test

```diff
@@ -0,0 +1,195 @@
+INSERT INTO edges VALUES  (0,4), (4,0),
+						  (0,3), (3,0),
+						  (1,4), (4,1),
+						  (3,4), (4,3),
+						  (2,5), (5,2);
+
+
+
+query II
+WITH RECURSIVE cc(node, comp) USING KEY (node) AS (
+	SELECT n.node, n.node AS comp
+	FROM nodes AS n
+		UNION
+	(SELECT DISTINCT ON (u.node) u.node, v.comp
+	FROM recurring_cc AS u, cc AS v, edges AS e
+	WHERE (e.here, e.there) = (u.node, v.node)
+	AND v.comp < u.comp
+	ORDER BY u.node asc, v.comp asc)
+)
+TABLE cc ORDER BY node;
+----
+0	0
+1	0
+2	2
+3	0
+4	0
+5	2
+6	6
+
+#######################
+# bellman-ford
+#######################
+statement ok
+CREATE OR REPLACE TABLE edges (here   char(2), -- source node
+  							   there  char(2), -- target node
+  							   length int,  -- edge weight
+  							   PRIMARY KEY (here, there));
+
+statement ok
+INSERT INTO edges(here, there, length) VALUES
+  ('v0', 'v1', 9),
+  ('v0', 'v2', 3),
+  ('v1', 'v2', 6),
+  ('v1', 'v4', 2),
+  ('v2', 'v1', 2),
+  ('v2', 'v3', 1),
+  ('v3', 'v2', 2),
+  ('v3', 'v4', 2);
+
+statement ok
+DROP TABLE nodes;
+
+statement ok
+CREATE OR REPLACE VIEW nodes(node) AS
+  SELECT e.here
+  FROM   edges AS e
+   UNION
+  SELECT e.there
+  FROM   edges AS e;
+
+query II
+WITH RECURSIVE bellman(knode, distance) USING KEY (knode) AS (
+  -- distance is 0 from source, ∞ from any other node
+  SELECT n.node AS knode,
+         CASE WHEN n.node = 'v0' THEN 0 ELSE 50000000 END AS distance
+  FROM   nodes AS n
+  UNION
+  (SELECT rec.knode AS knode, n.distance + e.length AS distance
+  FROM   bellman AS n, edges AS e, recurring_bellman AS rec
+  WHERE  (e.here, e.there) = (n.knode, rec.knode)
+  AND    rec.distance > n.distance + e.length
+  ORDER BY distance DESC)
+)
+SELECT n.knode || CASE WHEN n.knode = 'v0' THEN ' (source)' ELSE '' END AS node,
+       n.distance
+FROM   bellman AS n
+ORDER BY n.knode;
+----
+v0 (source)	0
+v1	5
+v2	3
+v3	4
+v4	6
+
+
+# more rows than STANDARD_CHUNK_SIZE
+query I
+WITH RECURSIVE tbl(n) USING KEY (n) AS (SELECT 1 UNION SELECT n+1 FROM tbl WHERE n < 3000) SELECT SUM(n) FROM tbl
+----
+4501500
+
+# You can use recursive CTE with and without a key in a query.
+query II
+WITH RECURSIVE tbl (a, b) USING KEY (a) AS (SELECT 1 , 1 UNION SELECT a, b+1 FROM tbl WHERE b < 3), tbl2(a,b) AS (SELECT 1, 1 UNION SELECT a, b+1 FROM tbl2 WHERE b < 3) SELECT * FROM tbl;
+----
+1	3
+
+query II
+WITH RECURSIVE tbl (a, b) USING KEY (a) AS (SELECT 1 , 1 UNION SELECT a, b+1 FROM tbl WHERE b < 3), tbl2(a,b) AS (SELECT 1, 1 UNION SELECT a, b+1 FROM tbl2 WHERE b < 3) SELECT * FROM tbl2;
+----
+1	1
+1	2
+1	3
+
+query II
+WITH RECURSIVE tbl2(a,b) AS (SELECT 1, 1 UNION SELECT a, b+1 FROM tbl2 WHERE b < 3), tbl (a, b) USING KEY (a) AS (SELECT 1 , 1 UNION SELECT a, b+1 FROM tbl WHERE b < 3) SELECT * FROM tbl2;
+----
+1	1
+1	2
+1	3
+
+query II
+WITH RECURSIVE tbl2(a,b) AS (SELECT 1, 1 UNION SELECT a, b+1 FROM tbl2 WHERE b < 3), tbl (a, b) USING KEY (a) AS (SELECT 1 , 1 UNION SELECT a, b+1 FROM tbl WHERE b < 3) SELECT * FROM tbl;
```

Comment:
Could we add some more tests with failure conditions?

* What if the key is not present in the table
* What if there are duplicate keys
* Given you are using an `expr_list_opt_comma`, what if the keys are arbitrary expressions (or perhaps this should be changed to `name_list_opt_comma` in the parser?)
* Can we add a test using these in subqueries?
* Can we add a test nesting these (similar to `test/sql/cte/test_nested_recursive_cte.test`)?
* Can we add a test nesting these alongside regular recursive CTEs?
* Can we add a test that uses `UNION ALL`?
* Can we add tests with more complex key types (e.g. string or list)?

Reply:
I will write more test, when the syntax is determined. :+1: